packages/admin/admin/src/mui/menu/Menu.tsx

```diff
@@ -1,13 +1,76 @@
-import { ComponentsOverrides, Drawer, DrawerProps, PaperProps, Theme } from "@mui/material";
-import { WithStyles, withStyles } from "@mui/styles";
+import Drawer, { DrawerProps } from "@mui/material/Drawer";
+import { PaperProps } from "@mui/material/Paper";
+import { ComponentsOverrides, css, styled, Theme, useThemeProps } from "@mui/material/styles";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 import { useHistory } from "react-router";
 
 import { MasterLayoutContext } from "../MasterLayoutContext";
 import { MenuContext } from "./Context";
-import { MenuClassKey, styles } from "./Menu.styles";
 
-export interface MenuProps {
+export type MenuClassKey = "drawer" | "permanent" | "temporary" | "open";
+
+type OwnerState = { open: boolean; variant: string };
+
+const StyledDrawer = styled(Drawer, {
+    name: "CometAdminMenu",
+    slot: "drawer",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [
+            styles.drawer,
+            ownerState.variant === "temporary" && styles.temporary,
+            ownerState.variant === "permanent" && ownerState.open && styles.permanent,
+        ];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ theme, ownerState }) => css`
+        .MuiDrawer-paper {
+            background-color: #fff;
+        }
+        .MuiPaper-root {
+            flex-grow: 1;
+            overflow-x: hidden;
+        }
+        .MuiDrawer-paperAnchorLeft {
+            border-right: none;
+        }
+
+        ${ownerState.variant === "permanent" &&
+        ownerState.open &&
+        css`
+            transition: ${theme.transitions.create("width", {
+                easing: theme.transitions.easing.sharp,
+                duration: theme.transitions.duration.enteringScreen,
+            })};
+
+            .MuiPaper-root {
+                transition: ${theme.transitions.create("margin", {
+                    easing: theme.transitions.easing.sharp,
+                    duration: theme.transitions.duration.enteringScreen,
+                })};
+            }
+        `}
+
+        ${ownerState.variant === "permanent" &&
+        !ownerState.open &&
+        css`
+            transition: ${theme.transitions.create("width", {
+                easing: theme.transitions.easing.easeOut,
+                duration: theme.transitions.duration.leavingScreen,
+            })};
+
+            .MuiPaper-root {
+                transition: ${theme.transitions.create("margin", {
+                    easing: theme.transitions.easing.easeOut,
+                    duration: theme.transitions.duration.enteringScreen,
+                })};
+                box-shadow: none;
+            }
+        `}
```

Comment:
Do the selectors still work correctly with `.MuiDrawer-paper` etc. even though this previously used the `[class*=...]` syntax?

Reply:
I couldn't find any case, that didn't work correctly. But to be sure, I changed the syntax back.